### ISSUE TITLE
[v1.7.0] Release Tracker

### ISSUE THREAD

**Issue by seemethere** [pytorch/pytorch#45592]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.7) for the 1.7.0 release.

Our plan from this point is roughly:

* **Phase 1 (until 10/13)**: work on finalizing the release branch
* **Phase 2 (after 10/13)**: perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 10/13):**

Only low-risk changes may be cherry-picked from master:

1.  Fixes to [regressions](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+regression%22) against PyTorch 1.0+
2.  Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3.  Fixes to new features being introduced in this release
4.  Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
5.  Test/CI fixes or improvements
6.  Documentation improvements
7.  Release branch specific changes (e.g. change version identifiers, remove/hide features that aren’t ready for release)

Any other change requires special dispensation from the release managers (currently @malfet, @seemethere, @gchanan).  If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain. 

**Phase 2 (after 10/13):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release).  Therefore, the only accepted changes are:

_Release-blocking_ critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

This will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1.  Ensure your PR has landed in master.  This does not apply for release-branch specific changes (see Phase 1 criteria).

2.  Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.7).  Note that you should use the normal github workflow for this; ghstack does not support release branches.

3.  Make a request below with the following template:

```
Link to landed master PR (if applicable):
Link to release branch PR:
Criteria Category:
```

4. Someone from the release team will reply with approved / denied or ask for more information 

5.  If approved, someone from the release team will merge your PR once the tests pass.  **NOTE: do not land the release branch PR yourself. Our normal landing tools (ghimport, phabricator, etc.) do not work on the release branch.**

**Comment by Krovatkin** [comment]
This change was slated for 1.7 but we narrowly missed the deadline.

https://github.com/pytorch/pytorch/pull/45591

**Comment by zou3519** [comment]
_Link to landed master PR (not necessary if release-specific change):_ n/a, release-specific change
_Link to release branch PR (create this yourself):_ https://github.com/pytorch/pytorch/pull/45571
_Criteria Category:_ Release branch specific changes: remove/hide features that aren’t ready for release

**Comment by Krovatkin** [comment]
master: 
**release branch:** 

Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45546
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45591
Criteria Category: n/a

**Comment by anjali411** [comment]
Link to landed master PR:
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45602
Criteria Category: silent correctness. Before this branch, the functions unaudited for complex backward returned incorrect result. After this PR, a `RuntimeError` is thrown in forward clearly stating that complex backward is not supported for the operator yet.

**Comment by suo** [comment]
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45628
Criteria category: Release branch specific changes: remove/hide features that aren't ready for release.

**Comment by jamesr66a** [comment]
https://github.com/pytorch/pytorch/pull/45631

**Comment by robieta** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45586
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45601
Criteria Category: special dispensation : Callgrind collection is an important aspect of the benchmarking utils beta launch. (CC @ngimel @ezyang @dzhulgakov) Original PR narrowly missed the cut, and had to be reverted and re-landed today due to a test breakage. Fortunately it is isolated to benchmark utils and as a result the interaction radius should be small.

**Comment by mrshenli** [comment]
Link to landed master PR: #45553
Link to release branch PR: #45738
Criteria Category: Documentation improvements: Advise users to use a more widely-adopted conda channel and pin the libuv version to the one that we have tested (1.39).

**Comment by mrshenli** [comment]
Link to landed master PR: #45554
Link to release branch PR: #45739
Criteria Category: Fixes to new features being introduced in this release: Make sure that users don't have to install libuv if they don't need the distributed package.

**Comment by QueensGambit** [comment]
Link to release branch PR: #45736
Criteria Category: Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries

**Comment by BowenBao** [comment]
Link to landed master PR: #44920
Link to release branch PR: #45755
Criteria Category: Fixes to new features being introduced in this release: Fix dim_param support for ONNX export with shape inference, along with fixes to shape propagation bugs in other code that are discovered. This PR is crucial for ONNX export in 1.7, unfortunately it missed the cut by a few hours.

**Comment by neginraoof** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/44693
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45756
Criteria Category: Fixes ONNX export of embedding bag operator with dynamically provided arguments: offsets and weights. Enables ONNX export of DLRM model. This PR is crucial for ONNX export in 1.7.

**Comment by osalpekar** [comment]
Link to landed master PR: #45543
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45758
Criteria Categrory: Documentation - PyTorch Distributed exposes a Python API for a distributed key-value store, but there were no public docs for this. This PR adds those public docs.

**Comment by bdhirsh** [comment]
Link to landed master PR (if applicable): #45587 and #45588
Link to release branch PR: #45759
Criteria Category: Fixed some incorrect changes to the documentation of `smooth_l1_loss`. Also preventing a new input (`beta`) in the backwards function from being defaulted, which is how we handle all other backwards function inputs and would otherwise require deprecating.

**Comment by xwang233** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45729
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45770
Criteria Category: Fixes to new features being introduced in this release. 

A user just reported seeing this https://github.com/pytorch/pytorch/issues/45769. It would be nice that we have this minor information printing bug be fixed in the release.

**Comment by anjali411** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45839
Criteria Category: Bug fixes for backward definition of complex functions

**Comment by gchanan** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45859
Criteria Category: Release branch specific changes -- makes the compatibility tests pass against 1.6.0.

**Comment by xta0** [comment]
Link to landed master PR:  #45651, #45677,  #45798 
Link to release branch PR: #45891 
Criteria Category:[iOS][CI]
1. Fix the iOS dev cert issue for CI jobs and nightlies.
2. Solve the XCode12 user issue - https://github.com/pytorch/pytorch/issues/44143

**Comment by vkuzo** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45848
Link to release branch PR: https://github.com/pytorch/pytorch/commit/ebe8b21b089ac75f053a67b5a980fce93a289d01
Criteria category: Documentation improvements - adds an API summary to the Quantization documentation

Working with @malfet to figure out how this ended up merged without a PR, I probably messed up a git step.

**Comment by xwang233** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45889 
Link to release branch PR: https://github.com/pytorch/pytorch/pull/45924
Criteria Category: 4. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries

This may improve the performance on Ampere GeForce 3080 and 3090.

**Comment by zasdfgbnm** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/46001, not landed yet
Link to release branch PR: Will create one after landed
Criteria Category: 2. Critical fixes for silent correctness https://github.com/pytorch/pytorch/issues/45724

**Comment by adampauls** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/43532
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46014
Criteria Category:4.  Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries

**Comment by t-vi** [comment]
Once it's landed, probably https://github.com/pytorch/pytorch/pull/46028

**Comment by malfet** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45914
Link to release brnch PR: https://github.com/pytorch/pytorch/pull/46039
Criteria Category: 3 or 5 - Speeds up checkout time

**Comment by anjali411** [comment]
Link to landed master PR: #45270 
Link to release PR: #46058 
Criteria: Documentation (Complex Autograd doc note)

**Comment by SplitInfinity** [comment]
Link to landed master PR: #45559 
Link to release PR: #46060
Criteria: Fixes to new features (JIT support for `torch.no_grad` being introduced in this release)

**Comment by pritamdamania87** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45642
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46070
Criteria Category: Critical fixes for: crashes: Using ProcessGroupNCCL without GPUs would result in a crash without this fix.

**Comment by wanchaol** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45871
Link to release PR: https://github.com/pytorch/pytorch/pull/46071

Criteria: critical fixes to new prototype features (distributed compilation concurrent compilation in multithreaded environment)

**Comment by rohan-varma** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45933
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46099
Criteria Category: Fixes to regressions against PyTorch 1.0+: Previously, DDP would report an error recommending users set `find_unused_parameters=True` when there is an error due to unused parameters in the model. This regressed due to https://github.com/pytorch/pytorch/pull/44798, and this cherry pick fixes that regression.

**Comment by neginraoof** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45874
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46100
Criteria Category: [ONNX] Improves error handling for ONNX exporter for scripting unsupported ops.

**Comment by albanD** [comment]
Link to landed master PR: None yet as there is some fb internal changes that are trickier to do
Link to release branch PR: #46103
Criteria Category: This op is not documented or tested so we shouldn't have it in the release.

**Comment by eellison** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/45857,  https://github.com/pytorch/pytorch/pull/45929
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46105
Criteria Category: critical fixes for silent correctness issue

**Comment by rohan-varma** [comment]
Link to landed master PR (if applicable): #46075
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46109
Criteria Category: Documentation improvements: We have implemented these APIs, but are still iterating on them. Thus, removes these from public docs for now. Will add them back when these APIs are part of a formal release.

**Comment by mrshenli** [comment]
Place holder for adding warning message for legacy APIs in `ProcessGroup::Work` cc @agolynski

**Comment by gchanan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45915
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46114
Criteria Category: Test/CI fixes or improvements

I don't think we _technically_ need this, but it's necessary for a backport of https://github.com/pytorch/pytorch/pull/46036 to apply cleanly.

**Comment by mruberry** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45802
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46177
Criteria Category: fix for silent correctness issue (https://github.com/pytorch/pytorch/issues/45721)

**Comment by malfet** [comment]
Link to landed master PR: #45004
Link to release branch PR: #46206
Criteria Category: Regression from previous release

**Comment by malfet** [comment]
Link to landed master PR: #46036
Link to release branch PR: #46207
Criteria Category: Fixes for silent correctness issue

**Comment by nickgg** [comment]
Link to landed master PR (if applicable): #46129
Link to release branch PR: #46214
Criteria Category: Correctness fixes for Half types in NNC based fuser (#45953

**Comment by vkuzo** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46077
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46217
Criteria Category: silent correctness (fixes numerics for NHWC in QNNPACK for tanh, hardsigmoid, sigmoid)

**Comment by gchanan** [comment]
Link to landed master PR (if applicable): N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46262
Criteria Category: Release branch specific changes

**Comment by anjali411** [comment]
Link to landed master: https://github.com/pytorch/pytorch/pull/46281
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46258
Criteria Category: Doc fix

**Comment by jeffdaily** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/46097
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46323
Criteria Category: Significant performance regression from previous release, ROCm-only

**Comment by gchanan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45782
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46333
Criteria Category: Test/CI fixes or improvements

should allow ROCm tests to pass.

**Comment by Flamefire** [comment]
Link to landed master PR (if applicable): #46186
Link to release branch PR: #46385
Criteria Category: Test/CI fixes or improvements

Fix for test_rnn in test/distributed/test_c10d_spawn.py

**Comment by Flamefire** [comment]
Link to landed master PR (if applicable): #46315
Link to release branch PR: #46388
Criteria Category: Test/CI fixes or improvements

Avoid error when using Python 3.9 (e.g. clang-tidy CI)

**Comment by albanD** [comment]
Link to landed master PR (if applicable): None
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46406
Criteria Category: 1.7 specific mitigation for https://github.com/pytorch/pytorch/issues/46242 that replaces the internal error by a helpful message.

**Comment by albanD** [comment]
Link to landed master PR (if applicable): #46397
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46408
Criteria Category: Fix wrong error message content for `torch.scatter`. It is a nice to have but not strictly needed.

**Comment by gchanan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/46335
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46418
Criteria Category: Test/CI fixes or improvements.  Should allow clang-tidy to pass on release branch.

**Comment by gchanan** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/45638
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46419
Criteria Category: Test/CI fixes or improvements. Should fix cuda fuser test failures on release branch.

**Comment by SplitInfinity** [comment]
Link to landed master PR: #45940
Link to release branch PR: #46422 
Criteria category: Fix for a crash in an uncommonly used user-facing code path in TorchScript

**Comment by gchanan** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46425
Criteria category: Release branch specific changes, fix CI for XLA.

**Comment by robieta** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46023
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46403
Criteria category: Special Dispensation several key tweaks to benchmark utils.

**Comment by ZolotukhinM** [comment]
Link to landed master PR: #46446
Link to release branch PR: #46453
Criteria category: fixes a crash

**Comment by gchanan** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46478
Criteria category: Release branch specific changes, fix CI for backwards compatibility

**Comment by Krovatkin** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46479
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46502
Criteria category: Critical fixes (increased memory consumption in profiling mode)

**Comment by ZolotukhinM** [comment]
Link to landed master PR: #46446, #46500, #46482
Link to release branch PR: #46513 (combines all three commits)
Criteria category: fixes incorrect behavior in fused aten::cat

**Comment by suo** [comment]
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46556
Link to master PR: https://github.com/pytorch/pytorch/pull/46556
Criteria category: fixes test breakage when built with BUILD_TEST=0 (which the RC builds do)

**Comment by albanD** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46673
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46718
Criteria category: Remove public functions that should not be public called `valgrind_*`.

**Comment by albanD** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46680
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46719
Criteria category: Remove public functions that should not be public for `__torch_function__` feature

**Comment by albanD** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46676
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46765
Criteria category: Remove public functions that should not be public `add_relu`.

**Comment by brianjo** [comment]
Link to release branch PR: #46845 
Criteria category: Documentation improvements.

**Comment by osalpekar** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/46856
Link to release branch PR: https://github.com/pytorch/pytorch/pull/46879
Criteria category: Docs fixes

**Comment by z-a-f** [comment]
Link to landed master PR: #46858 
Link to release branch PR: #46890
Criteria Category: Documentation improvement: Quantization docs had a specific version to describe unsupported features. Changed to more general wording.

**Comment by seemethere** [comment]
Hey everyone, 1.7.0 has been released so I'm going to close this.

Thank you everyone for the work towards getting v1.7.0 out of the door!

**Comment by z-a-f** [comment]
> Link to landed master PR: #46858
> Link to release branch PR: #46890
> Criteria Category: Documentation improvement: Quantization docs had a specific version to describe unsupported features. Changed to more general wording.

@seemethere If this PR was not merged, is there still a way to add it?
